xwp/block-extend
# Support Composer 2

* Though this plugin will be archived, this PR might help in the meantime 
* Composer 1 is [EOL](https://blog.packagist.com/deprecating-composer-1-support/)
* Someone mentioned that this plugin doesn't work with Composer 2:

```sh
\W $composer install        
The "dealerdirect/phpcodesniffer-composer-installer" plugin was skipped because it requires a Plugin API version ("^1.0") that does not match your Composer installation ("2.0.0"). You may need to run composer update with the "--no-plugins" option.
The "composer/installers" plugin was skipped because it requires a Plugin API version ("^1.0") that does not match your Composer installation ("2.0.0"). You may need to run composer update with the "--no-plugins" option.
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - composer/installers is locked to version v1.7.0 and an update of this package was not requested.
    - composer/installers v1.7.0 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
  Problem 2
    - dealerdirect/phpcodesniffer-composer-installer is locked to version v0.6.2 and an update of this package was not requested.
    - dealerdirect/phpcodesniffer-composer-installer v0.6.2 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.

You are using Composer 2, which some of your plugins seem to be incompatible with. Make sure you update your plugins or report a plugin-issue to ask them to support Composer 2.
\W $b
```

* It'd be nice if `composer.lock` was a smaller diff, but I couldn't figure out how to make it simpler

## Testing Steps
1. `composer self-update --2`
2. This will ensure Composer is version 2
3. `composer install`
4. Expected: there's no error
5. `composer test` 
6. Expected: tests pas
7. `composer lint` 
8. Expected: linting passes
